### ISSUE TITLE
github: badges should contain no caching headers

### ISSUE THREAD

**Issue by krzysztof** [zenodo/zenodo#912]
1. Create a GitHub release, insert the badge in README
1. Create another release
1. **Badge still renders the old image, although it properly redirects to the new release**
1. Refreshing browser cache does not work (nor does changing the browser), since the images are hosted (and cached) on GitHub side once the badge is added.

This is a Travis badge SVG response headers:
```
> $ curl https://travis-ci.org/zenodo/zenodo.svg\?branch\=master -L -I                                                                                                                                                                                                                                                        
HTTP/1.1 301 Moved Permanently
Server: Cowboy
Date: Mon, 28 Nov 2016 13:11:35 GMT
Connection: keep-alive
Content-Type: text/html;charset=utf-8
Location: https://api.travis-ci.org/zenodo/zenodo.svg?branch=master
X-Xss-Protection: 1; mode=block
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
Content-Length: 0
Via: 1.1 vegur

HTTP/1.1 200 OK
Connection: keep-alive
Server: nginx
Date: Mon, 28 Nov 2016 13:11:36 GMT
Content-Type: image/svg+xml
Content-Length: 726
Access-Control-Allow-Origin: *
Access-Control-Allow-Credentials: true
Access-Control-Expose-Headers: Content-Type, Cache-Control, Expires, Etag, Last-Modified
Strict-Transport-Security: max-age=31536000
X-Endpoint: Travis::Api::App::Endpoint::Repos
X-Pattern: /:owner_name/:name
X-Oauth-Scopes: public
X-Accepted-Oauth-Scopes: public
Vary: Accept,Accept-Encoding
Cache-Control: no-cache
Etag: "a9e445ad2237440326816952c70dba79"
Pragma: no-cache
Expires: Mon, 28 Nov 2016 13:11:36 GMT
Content-Disposition: inline; filename="failing.svg"
Last-Modified: Thu, 24 Nov 2016 15:27:27 GMT
X-Content-Digest: 922c9cc746e8bc766b6acfed45d83e014df8a13e
Age: 0
X-Rack-Cache: miss, store
Via: 1.1 vegur
```

while this is from our badge:
```
$ curl https://zenodo.org/badge/6365411.svg -L -I                                                                                                                                                                                                                                                                           
HTTP/1.1 302 FOUND
Server: nginx/1.10.2
Date: Mon, 28 Nov 2016 13:06:05 GMT
Content-Type: text/html; charset=utf-8
Content-Length: 279
Location: https://zenodo.org/badge/doi/10.5281/zenodo.178482.svg
X-Sentry-ID: None
Strict-Transport-Security: max-age=31536000

HTTP/1.1 200 OK
Server: nginx/1.10.2
Date: Mon, 28 Nov 2016 13:06:05 GMT
Content-Type: image/svg+xml
Content-Length: 1189
X-Sentry-ID: None
Strict-Transport-Security: max-age=31536000
```

In our badge responses we should include headers such as ``Cache-Control``, ``Etag``, ``Pragma``, ``Expires``, ``Last-Modified`` so they are correctly not cached on GitHub. GitHub is hosting and caching the images on their side (probably copying our original header settings), hence the current badges will not be automatically refreshed if we start to cache information in new headers.

**Comment by slint** [comment]
This could be implemented by extending the API the `invenio-formatter`'s [badge view](https://github.com/inveniosoftware/invenio-formatter/blob/master/invenio_formatter/views.py#L47-L63), since all kinds of badges could benefit from some sort of cache control (eg. Versioned DOI badges in the future). So additionally to badge `title` and `value`, you could pass also `etag`, `last_modified`, etc.

**Comment by slint** [comment]
Closed in favor of #1164
Moved to https://github.com/inveniosoftware/invenio-formatter/issues/54
